File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        i = 0
        j = 0
        m = 0
        l = 0
        myset = []
        while i<len(s):
            if j<len(s) and (s[j] not in myset):
                myset.append(s[j])
                j+=1
                l+=1
                m = max(m,l)
            else:
                myset = []
                i+=1
                j = i
                l = 0
        return m
        

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
import java.util.*;
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int i = 0,j=0,len=0,max=0,n=s.length();
        Set<Character> set = new HashSet<>();
        while(i<n){
            if(j<s.length() && !set.contains(s.charAt(j))){
                set.add(s.charAt(j));
                j++;
                len++;
                max = Math.max(len,max);
            }
            else{
                set.clear();
                i++;
                j=i;
                len=0;
            }
           
        }
         return max;
    }
}